report/4.Solution/SemanticsDescription.tex

```diff
@@ -0,0 +1,748 @@
+\begin{center}
+	$env_V,sto \vdash e \rightarrow v$
+\end{center} 
+
+\subsection{Big-Step Semantic for Expressions}
+An expression is any syntactic component that evaluates to a value.
+In HCL most expressions behave the same, but a few transition-systems uses special notation.
+When this is the case, it will be made apparent before that transition-system.
+
+\textbf{\large{General Rules for Expressions}} \\
+All expression in the abstract syntax have two rewriting rules in common.
+The variable rule and the parenthesis rule.
+To avoid the possibility of cluttering the section, these rules are described below, using the metavariable $e \in Exp$, where the syntax category $Exp$ is the set of all expressions.
+
+\textbf{[$Parent_{BSS}$]}
+\begin{center}
+	\begin{math}
+	\cfrac
+		{env_V,\ sto \vdash e \rightarrow_e v}
+		{env_V,\ sto \vdash (e) \rightarrow_e v}
+	\end{math}
+\end{center}
+
+\textbf{[$Var_{BSS}$]}
+\begin{center}
+	\begin{math}
+	env_V,\ sto \vdash x \rightarrow_e v
+	\texttt{ if } env_V\ x = l
+	\texttt{ and } sto\ l = v
+	\end{math}
+\end{center}
+
+\textbf{\large{Arithmetic Expressions}}\\
+Arithmetic expressions returns a value $v$, where $v \in \mathbb{R}$ or $v \in \mathbb{R}$.
+The transitions-system for $A_{exp}$ is $(A_{exp} \cup D, \rightarrow_a, D)$, 
+where $D = \mathbb{Z} \cup \mathbb{R}$.
+The transitions can be seen below.
+
+\textbf{[$Add_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	\cfrac
+		{env_V,\ sto \vdash a_1 \rightarrow_a v_1 \quad env_V, sto \vdash a_2 \rightarrow_a v_2}
+		{env_V,\ sto \vdash a_1 + a_2 \rightarrow_a v}
+	\end{math}
+	
+	
+	\texttt{where} $v = v_1 + v_2$
+\end{center}
+
+\textbf{[$Minus_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	\cfrac
+		{env_V,\ sto \vdash a_1 \rightarrow_a v_1 \quad env_V, sto \vdash a_2 \rightarrow_a v_2}
+		{env_V,\ sto \vdash a_1 - a_2 \rightarrow_a v}
+	\end{math}
+	
+	
+	\texttt{where} $v = v_1 - v_2$
+\end{center}
+
+\textbf{[$Mult_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	\cfrac
+	{env_V,\ sto \vdash a_1 \rightarrow_a v_1 \quad env_V, sto \vdash a_2 \rightarrow_a v_2}
+	{env_V,\ sto \vdash a_1 * a_2 \rightarrow_a v}
+	\end{math}
+	
+	
+	\texttt{where} $v = v_1 \cdot v_2$
+\end{center}
+
+\textbf{[$Div_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	\cfrac
+	{env_V,\ sto \vdash a_1 \rightarrow_a v_1 \quad env_V, sto \vdash a_2 \rightarrow_a v_2}
+	{env_V,\ sto \vdash a_1 / a_2 \rightarrow_a v}
+	\end{math}
+	
+	
+	\texttt{where} $v = v_1 / v_2$
+\end{center}
+
+\textbf{[$Num_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		env_V,\ sto \vdash n \rightarrow_a v
+	\end{math}
+	\texttt{ if } $\mathbb{N}[n] = v$
+\end{center}
+
+\textbf{\large{Boolean Expressions}}\\
+Boolean expressions returns a value $v$, where $v \in \{True, False\}$.
+The transition-system for $B_{exp}$ is $(B_{exp} \cup \{True, False,\} \rightarrow_b, \{True, False\})$.
+The transition can be seen below.
+
+\textbf{[$EqualTrue_{BSS}$]}\\
```

Comment:
define what equals mean. is it the value? is the the address? what defines two values being the same? is "hello" and "hello " the same? (it's not)

Reply:
Done.
Fixed